packages/uploadthing/src/client.ts

```diff
@@ -259,12 +260,68 @@ export const DANGEROUS__uploadFiles = async <TRouter extends FileRouter>(
       fileUrl: genUrl,
       url: genUrl,
     };
+
+    console.log("fileData", fileData);
+
+    if (!fileData) {
+      console.log("[UT] Could not receive file data from UT");
+    } else {
+      console.log("[UT] Trying to simulate the webhook");
+
+      await conditionalDevServer(fileData.fileData, presigned.key);
+    }
+
     return ret;
   });
 
   return Promise.all(fileUploadPromises);
 };
 
+const conditionalDevServer = async (file: FileData, fileKey: string) => {
+  if (process.env.NODE_ENV !== "development") return;
+
+  let callbackUrl = file.callbackUrl + `?slug=${file.callbackSlug}`;
+  if (!callbackUrl.startsWith("http")) callbackUrl = "http://" + callbackUrl;
+
+  console.log("[UT] SIMULATING FILE UPLOAD WEBHOOK CALLBACK", callbackUrl);
+
+  const response = await fetch(callbackUrl, {
+    method: "POST",
+    body: JSON.stringify({
+      status: "uploaded",
+      metadata: JSON.parse(file.metadata ?? "{}") as FileData["metadata"],
+      file: {
+        url: `https://uploadthing.com/f/${encodeURIComponent(fileKey)}`,
+        key: fileKey,
+        name: file.fileName,
+        size: file.fileSize,
+      },
+    }),
+    headers: {
+      "uploadthing-hook": "callback",
+    },
+  });
+
+  if (isValidResponse(response)) {
+    console.log("[UT] Successfully simulated callback for file", fileKey);
+  } else {
+    console.error(
```

Comment:
Should this be throwing instead?

Reply:
Hm, throwed exception may affect user code execution. Since this hook simulation happens in dev mode only, I don't think user has to make any guards for it on client side. I mean this code will become useless in production as it will never be called